app/src/main/java/com/kickstarter/viewmodels/PledgeFragmentViewModel.kt

```diff
@@ -313,12 +319,22 @@ interface PledgeFragmentViewModel {
                     .filter { ObjectUtils.isNull(it.first) || it.first.isEmpty() }
                     .compose<Pair<Pair<List<ShippingRule>, Reward>, Double>>(combineLatestPair(rewardAmountPlusAdditional))
 
-            val initialTotalAmount = rulesAndRewardAndAdditional
+            val initialTotalAmountWithReward = rulesAndRewardAndAdditional
                     .map { it.second }
                     .compose<Pair<Double, Project>>(combineLatestPair(project))
                     .map<SpannableString> { this.ksCurrency.formatWithProjectCurrency(it.first, it.second, RoundingMode.UP, 2) }
                     .compose(bindToLifecycle())
 
+            val initialTotalAmountNoReward = rewardAmountPlusAdditional
```

Comment:
Since this is coming up again; instead of calling ```.map<SpannableString> { this.ksCurrency.formatWithProjectCurrency(it.first, it.second, RoundingMode.UP, 2) }``` 3 times, you can do this:
```
            val initialTotalAmountWithReward = rulesAndRewardAndAdditional
                    .map { it.second }
                    .compose<Pair<Double, Project>>(combineLatestPair(project))

            val initialTotalAmountNoReward = rewardAmountPlusAdditional
                    .compose<Pair<Double, Reward>>(combineLatestPair(reward))
                    .filter { RewardUtils.isNoReward(it.second) }
                    .map { it.first }
                    .compose<Pair<Double, Project>>(combineLatestPair(project))

            val initialTotalAmount = Observable.merge(initialTotalAmountNoReward, initialTotalAmountWithReward)

            val totalWithShippingRule = rewardAmountPlusAdditional
                    .compose<Pair<Double, Double>>(combineLatestPair(shippingAmount))
                    .map { it.first + it.second }
                    .compose<Pair<Double, Project>>(combineLatestPair(project))

            Observable.merge(initialTotalAmount, totalWithShippingRule)
                    .map<SpannableString> { this.ksCurrency.formatWithProjectCurrency(it.first, it.second, RoundingMode.UP, 2) }
                    .compose(bindToLifecycle())
                    .subscribe(this.totalAmount)
```
`totalAmount` is what cares about the `SpannableString` so that's when we can map to get it. Also we only need to call `.compose(bindToLifecycle())` before subscribing.